### PR TITLE
[CNXC-291] Analysis Created Pluralization

### PR DESCRIPTION
**Pull request by colinywu**
### Bug
When an analysis is ran 1 day ago, in the Past Analysis Table under the 'Analysis Created' column it says '1 days ago'

### Expected Behaviour
The value of the cell should be '1 day ago'

### Implementation
Added checking if the elapsed time since the analysis is greater than 1 day, and assigned string accordingly

### CODE REVIEW

**Review by boris-nguyen — CHANGES_REQUESTED** [review]
I would suggest using `moment.js` to get the relative time.

Firstly, we are already using `moment.js` in other places to format dates, so using moment for all time formatting would keep it consistent. Secondly, relative time in moment supports more relative date ranges such as "years ago"; in our implementation it would look like "132 weeks ago" as an example. Thirdly, using moment would help us avoid bugs and edge cases that were not considered during development of `modifyDatetime`. Lastly, we won't be reinventing a wheel.

@janakitti @colinywu let me know what you think about replacing `modifyDatetime()` logic with  `moment(<timestamp>).fromNow()`.

If we decide to leave the original logic, then I would highly suggest adding "year(s) ago" relative time.

**Comment by janakitti** [comment]
> I would suggest using `moment.js` to get the relative time.
> 
> Firstly, we are already using `moment.js` in other places to format dates, so using moment for all time formatting would keep it consistent. Secondly, relative time in moment supports more relative date ranges such as "years ago"; in our implementation it would look like "132 weeks ago" as an example. Thirdly, using moment would help us avoid bugs and edge cases that were not considered during development of `modifyDatetime`. Lastly, we won't be reinventing a wheel.
> 
> @janakitti @colinywu let me know what you think about replacing `modifyDatetime()` logic with `moment(<timestamp>).fromNow()`.
> 
> If we decide to leave the original logic, then I would highly suggest adding "year(s) ago" relative time.

I agree that we should be using a library to handle this formatting. However. Moment.js is now considered a legacy project and they don't advise using it in new projects (and it's apparently also pretty bulky). COVID-Net currently only uses it in a few places. Would it be a good idea now to switch to another library (like Luxon)?

**Comment by boris-nguyen** [comment]
> > I would suggest using `moment.js` to get the relative time.
> > Firstly, we are already using `moment.js` in other places to format dates, so using moment for all time formatting would keep it consistent. Secondly, relative time in moment supports more relative date ranges such as "years ago"; in our implementation it would look like "132 weeks ago" as an example. Thirdly, using moment would help us avoid bugs and edge cases that were not considered during development of `modifyDatetime`. Lastly, we won't be reinventing a wheel.
> > @janakitti @colinywu let me know what you think about replacing `modifyDatetime()` logic with `moment(<timestamp>).fromNow()`.
> > If we decide to leave the original logic, then I would highly suggest adding "year(s) ago" relative time.
> 
> @boris-nguyen @colinywu I agree that we should be using a library to handle this formatting. However. Moment.js is now considered a legacy project and they don't advise using it in new projects (and it's apparently also pretty bulky). COVID-Net currently only uses it in a few places. Would it be a good idea now to switch to another library (like Luxon)?

Nice catch @janakitti, didn't realize it became a legacy project a couple of months ago. Seems like Luxon is indeed the way to go.
https://momentjs.com/docs/#/-project-status/
https://terodox.tech/migrating-away-from-momentjs-part1/

Here's what I'm thinking:
1) Do a quick research about what is the best option for migrating away from moment.js
2) Estimate how much effort it would require to migrate away from moment.js
3) If it requires a lot of effort, create a separate Jira ticket to tackle the migration and in meantime just add "years ago" handling. If it does not require a lot of effort, then migrate away from moment.js.

@janakitti @colinywu Let me know what is your opinion about this.

**Review by janakitti — APPROVED** [review]
Great work, @colinywu!

File: src/shared/utils.tsx [hunk 1]
```diff
@@ -1,37 +1,21 @@
-import moment from "moment";
+import { DateTime } from "luxon";
 
 export const formatDate = (dateStr: string): string => {
-  return moment(dateStr).format("YYYY MM DD");
+  return DateTime.fromISO(dateStr).toFormat('yyyy MM dd');
```
Comment (boris-nguyen) [inline src/shared/utils.tsx:4]:
```suggestion
  return DateTime.fromISO(dateStr).toFormat("yyyy MM dd");
```

File: src/shared/utils.tsx [hunk 2]
```diff
@@ -1,37 +1,21 @@
...
 }
 
 export const calculatePatientAge = (patientDOB: string): number => {
-  return moment().diff(patientDOB, 'year');
+  return Math.trunc(-DateTime.fromISO(patientDOB).diffNow('years').years);
 }
 
 export const formatTime = (oldDay: string): string => {
-  return oldDay.split('.')[0]
+  return oldDay.split('.')[0];
```
Comment (boris-nguyen) [inline src/shared/utils.tsx:16]:
```suggestion
  return oldDay.split(".")[0];
```

File: src/shared/utils.tsx [hunk 3]
```diff
@@ -1,37 +1,21 @@
...
+  return DateTime.fromISO(dateStr).toFormat("yyyy MM dd");
 }
 
 export const formatGender = (gender: string): string => {
   return gender.includes("F") ? "Female" : "Male";
 }
 
 export const calculatePatientAge = (patientDOB: string): number => {
-  return moment().diff(patientDOB, 'year');
+  return Math.trunc(-DateTime.fromISO(patientDOB).diffNow('years').years);
```
Comment (boris-nguyen) [inline src/shared/utils.tsx:12]:
```suggestion
  return Math.trunc(-DateTime.fromISO(patientDOB).diffNow("years").years);
```

File: src/shared/utils.tsx [hunk 4]
```diff
@@ -1,37 +1,21 @@
...
-    rvtVal = `${diff} ${description}`
-  } else if (diff / 7 >= 1) {
-    description = diff / 7 >= 2 ? "weeks ago" : "week ago"
-    diff = Math.floor(diff / 7)
-    rvtVal = `${diff} ${description}`
-  } else if (diff < 1) {
-    rvtVal = oldDay.split('T')[1].split('.')[0]
-  }
-  return rvtVal
+  return (DateTime.fromISO(oldDay).toRelativeCalendar() || "").split(' ').map((date: string) => date.charAt(0).toUpperCase() + date.slice(1)).join(' ');
```
Comment (boris-nguyen) [inline src/shared/utils.tsx:20]:
```suggestion
  return (DateTime.fromISO(oldDay).toRelativeCalendar() || "").split(" ").map((date: string) => date.charAt(0).toUpperCase() + date.slice(1)).join(" ");
```

**Review by boris-nguyen — APPROVED** [review]
Good work @colinywu

**Pull request merged**